jobfunnel/config/cli.py

```diff
@@ -0,0 +1,393 @@
+    )
+
+    parser.add_argument(
+        '-delay-max',
+        dest='delay_max_duration',
+        type=float,
+        help='Set delay seconds for certain get requests.'
+    )
+
+    parser.add_argument(
+        '-delay-min',
+        dest='delay_min_duration',
+        type=float,
+        help='Set lower bound value for delay for certain get requests.'
+    )
+
+    parser.add_argument(
+        '-delay-algorithm',
+        choices=[a.name for a in DelayAlgorithm],
+        help='Select a function to calculate delay times with.'
+    )
+
+    return parser.parse_args()
+
+
+def config_builder(args: argparse.Namespace) -> JobFunnelConfigManager:
+    """Parse the JobFunnel configuration settings into a JobFunnelConfigManager.
+
+        args [argparse.Namespace]: cli arguments from argparser
+    """
+    # Init and pop args that are cli-only and not in our schema
+    args_dict = vars(args)
+    settings_yaml_file = args_dict.pop('settings_yaml_file')
+    output_folder = args_dict.pop('output_folder')
+    args_dict.pop('do_recovery_mode')  # NOTE: this is handled in __main__
+
+    # Load config dict from the YAML if passed
+    config = {'search': {}, 'delay': {}, 'proxy': {}}
+    if settings_yaml_file:
+        config.update(
+            yaml.load(open(settings_yaml_file, 'r'), Loader=yaml.FullLoader)
+        )
+
+        # TODO: need a way to handle injecting defaults with YAML but also
+        # without an incomplete set of arguments here. Cerberus should do this
+        # if we further break down config into sub-configs.
+        missing_attrs = []  # type: List[str]
+        for attr in ['master_csv_file', 'block_list_file', 'cache_folder',
+                     'duplicates_list_file']:
+            if attr not in config:
+                missing_attrs.append(attr)
+        if missing_attrs:
+            raise ValueError(
+                f"Passed YAML {settings_yaml_file} fields are missing or "
+                f"invalid: {missing_attrs}"
+            )
+
+    # Handle output_folder argument which is a shortcut to specifying all paths
+    user_passed_paths = bool(
+        (
+            args_dict['master_csv_file'] and args_dict['block_list_file'] and
+            args_dict['duplicates_list_file'] and args_dict['cache_folder']
+        ) or (
+            config.get('master_csv_file') and config.get('block_list_file') and
+            config.get('duplicates_list_file') and config.get('cache_folder')
+        )
+    )
+    if output_folder == DEFAULT_OUTPUT_DIRECTORY and not user_passed_paths:
+
+        # We are using all defaults only (no -s or paths passed)
+        config['master_csv_file'] = DEFAULT_MASTER_CSV_FILE
+        config['block_list_file'] = DEFAULT_BLOCK_LIST_FILE
+        config['duplicates_list_file'] = DEFAULT_DUPLICATES_FILE
+        config['cache_folder'] = DEFAULT_CACHE_DIRECTORY
+
+    elif output_folder != DEFAULT_OUTPUT_DIRECTORY and user_passed_paths:
```

Comment:
Currently this logic check will not check for an existing _settings.yaml_ file which leads to storing 'None' within the _settings_yaml_file_ variable and creating an incomplete config dictionary; therefore, breaking at config validation.

Reply:
thanks for finding this as well, will revise.

Reply:
Ok, this should be fixed now